packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/__tests__/quanticDocumentSuggestion.test.js

```diff
@@ -0,0 +1,274 @@
+        'mockedFetchDocumentSuggestions'
+      );
+    });
+  });
+
+  describe('with suggestions', () => {
+    beforeEach(() => {
+      documentSuggestionListMock.state.documents = [
+        {
+          uniqueId: 'ego',
+          fields: {
+            uri: 'weed eater',
+          },
+        },
+        {
+          uniqueId: 'stihl',
+          fields: {
+            uri: 'chainsaw',
+          },
+        },
+      ];
+    });
+
+    it('should render the component and all parts with default options', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      checkElement(element, selectors.accordion, false);
+      checkElement(element, selectors.noSuggestions, true);
+      checkElement(element, selectors.componentError, false);
+
+      await updateSuggestions();
+
+      expect(engineMock.dispatch).not.toHaveBeenCalled();
+      expect(headlessLoaderMock.registerComponentForInit).toHaveBeenCalled();
+
+      checkElement(element, selectors.noSuggestions, false);
+      checkElement(element, selectors.container, true);
+      checkElement(element, selectors.accordion, true);
+      const sections = element.shadowRoot.querySelectorAll(
+        selectors.accordionSection
+      );
+      expect(sections.length).toBe(2);
+      expect(sections[0].getAttribute('data-id')).toBe('ego');
+      checkElement(element, selectors.quickview, true);
+    });
+
+    it('should not render quick view button when withoutQuickview is set to true', async () => {
+      const element = createTestComponent({
+        withoutQuickview: true,
+        fetchOnInit: true,
+      });
+      await flushPromises();
+
+      expect(element.shadowRoot.querySelector(selectors.quickview)).toBeFalsy();
+    });
+
+    it('should log a suggestion click if not opened', async () => {
+      const element = createTestComponent({
+        fetchOnInit: true,
+      });
+      await updateSuggestions();
+
+      const accordion = element.shadowRoot.querySelector(selectors.accordion);
+      const sections = element.shadowRoot.querySelectorAll(
+        selectors.accordionSection
+      );
+
+      // Should not log click when opening first
+      sections[0].click();
+      expect(engineMock.dispatch).not.toHaveBeenCalledWith(
+        'mockedLogDocumentSuggestionClick'
+      );
+
+      accordion.activeSectionName = sections[1].name;
+      sections[1].click();
+      expect(engineMock.dispatch).toHaveBeenCalledWith([
+        'mockedLogDocumentSuggestionClick',
+        sections[1].name,
+      ]);
+    });
+
+    it('should handle document rating event', async () => {
+      const element = createTestComponent({
+        fetchOnInit: true,
+      });
+      await updateSuggestions();
+
+      const customEvent = new CustomEvent('quantic__rating', {
+        detail: {id: 'mastercraft', score: 2},
+      });
+      element.shadowRoot.dispatchEvent(customEvent);
+      expect(engineMock.dispatch).toHaveBeenCalledWith([
```

Comment:
wherever we use `toHaveBeenCalledWith ` to check the payload of functions called we also need to call `toHaveBeenCalledTimes` to make sure they are called the number of times we expect.